gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -152,11 +166,9 @@ type stdlibCache struct {
 }
 
 var (
-	cachedStdlibOnce sync.Once
-	cachedStdlib     stdlibCache
-	// XXX: this is shared across different goroutines, in txtar tests.
-	// need to find a better way, or put a lock.
-	sharedTypeCheckCache gno.TypeCheckCache
+	cachedStdlibOnce         sync.Once
+	cachedStdlib             stdlibCache
+	cachedInitTypeCheckCache gno.TypeCheckCache
 )
 
 // LoadStdlib loads the Gno standard library into the given store.
```

Comment:
please update the comment to clearly explain how the cache works and what purpose it serves.